### PR TITLE
Small code cleanup - clean version

### PR DESCRIPTION
**Pull request by Woazboat**
Was #1917 
New changes from old pull request:
Cleaned up commits
Initialize Tracer::m_NormalTable on first use
clang warning overrides

### CODE REVIEW

File: src/Tracer.cpp [hunk 1]
```diff
@@ -241,7 +264,7 @@ bool cTracer::Trace(const Vector3f & a_Start, const Vector3f & a_Direction, int
 			int Normal = GetHitNormal(a_Start, End, pos);
 			if (Normal > 0)
 			{
-				HitNormal = m_NormalTable[Normal-1];
+				HitNormal = m_NormalTable()[Normal-1];
```
Comment (madmaxoft) [inline src/Tracer.cpp]: Add spaces around the `-`

File: src/Tracer.h [hunk 2]
```diff
@@ -61,10 +63,11 @@ class cTracer
 	/// Return 1 through 6 for the following block faces, repectively: -x, -z, x, z, y, -y
 	int GetHitNormal( const Vector3f & start, const Vector3f & end, const Vector3i &  a_BlockPos);
 
-	float SigNum( float a_Num);
+	/// Signum function
+	int SigNum( float a_Num);
 	cWorld* m_World;
 
-	Vector3f m_NormalTable[6];
+	static const std::array<const Vector3f, 6>& m_NormalTable(void);
```
Comment (madmaxoft) [inline src/Tracer.h]: Add a space in front of the `&`, too

File: src/Vector3.h [hunk 3]
```diff
@@ -78,6 +78,20 @@ class Vector3
 		);
 	}
 
+	inline bool HasNonZeroLength(void) const
+	{
+#ifndef __GNUC__
+#pragma clang diagnostics push
+#pragma clang diagnostics ignored "-Wfloat-equal"
+#endif
```
Comment (madmaxoft) [inline src/Vector3.h]: Preprocessor blocks should use regular indent

**Comment by bearbin** [comment]
Is this good to merge?

**Comment by Woazboat** [comment]
It is from my side. Looks like the coverage checker somehow got stuck though.

**Comment by SafwatHalaby** [comment]
Coverity.io is also stuck in #1968.

**Pull request merged**